### PR TITLE
Adding offset, min, and max options to allow for complete collection on keywords using the search function

### PR DESCRIPTION
**Pull request by patrick-lee-warren**
**This is literally by first pull request ever, so be gentle.**

Usage is something like this:

```
min=108789781130656241
gap=100000000000
cycles=100
term='Russia'

search_count=30
page=0
status_list=[]

for cycle in range(0,cycles):
    search_count=30
    page=0
    while search_count>0:
        temp=test_api.search('statuses', term, 40, 4,40*page, str(min+cycle*gap),str(min+(cycle+1)*gap) )['statuses']
        status_list=status_list+temp
        search_count=len(temp)
        page+=1

```

### CODE REVIEW

File: truthbrush/api.py [hunk 1]
```diff
@@ -123,20 +123,41 @@ def search(
...
+                    q=query,
+                    resolve=resolve,
+                    limit=limit,
+                    type=searchtype,
+                    offset=offset,
+                    min_id=min_id,
+                    
+                ),
+            )
+        else:
```
Comment (KonradIT) [inline truthbrush/api.py:146]: I'd remove the else to fit the

```
if y:
    return a
return b
```

pattern.

Comment (patrick-lee-warren) [reply truthbrush/api.py:146]: Is that better practice? I just kludge things. Happy to do it.

Comment (patrick-lee-warren) [reply truthbrush/api.py]: Done

**Comment by lxcode** [comment]
This looks like it should work as the basic mechanism — we'd want to have the pagination logic be similar to what's currently used for _get_paginated.

I know it's not part of the PR, but can you describe how you came up with the value for gap?

**Comment by patrick-lee-warren** [comment]
> 
I'm still playing with it. It really depends on how prominent is the keyword you are using for your search.

**Comment by lxcode** [comment]
Any problem with just letting the user select the max and min IDs themselves and just stop anytime we find an ID greater than the max? Seems easier than futzing with the offsets, and if you're using the min/max IDs to backfill data, you should already know what you want.

**Comment by patrick-lee-warren** [comment]
There is a max offset of 10k.

**Comment by lxcode** [comment]
I mean the min_id and max_id parameters, not the offset.

**Comment by patrick-lee-warren** [comment]
Now I'm confused. The push does let the user set min and max. The code I posted above just showed how I was using the new updated search function to collect all the Truths that satisfy my search criteria.

**Comment by lxcode** [comment]
Ok — I thought that was something you were going to eventually going to work into the pagination logic — if not, then no matter.

**Comment by lxcode** [comment]
Merging: we should still put pagination support in there, but it doesn't cause any regressions and gives us a framework for some future additions. Thanks!

**Comment by lxcode** [comment]
(cleaning up previous discussion to prevent confusion, sorry)

Ok, pagination logic is added now, let me know if you find any problems with it.

**Pull request merged**